File: Kotlin/src/compare.kt
//spread operator (vararg in kotlin) is unsupported in functional style in kotlin
fun <K,V> compare(first_values: Map<K,V>, vararg parameters_LIST: Map<K,V>): Boolean {
    val second_values = parameters_LIST[0]
    var a = first_values.keys
    var b = second_values.keys
    if(a.joinToString("-") != b.joinToString("-") ) return false
    var e = true
    for (c in a){
        if (first_values[c] == second_values[c]) e = e && true
        else { e = e && false }
    }
    return e
}
fun main(){
    val result = compare(mapOf('a' to 1, 'c' to "hello"), mapOf('a' to 1, 'c' to "hello"))
    println(result)
}

File: NotFormatted/compare.js
// Compare two dictionaries

let compare=(
  (first_values,...parameters_LIST) => {
   const second_values = parameters_LIST[0]
   let a = Object.keys(first_values)
   let b = Object.keys(second_values)
   if (a.join('-') !== b.join('-')) return false;
   let e = true;
   for (c of a) {
    if (first_values[c] === second_values[c]) e = e && true
   else { e = e && false }
   }
  return e;
  }
)

require('../Tests/compare.js')(compare);
